code/modules/mob/living/simple_animal/hostile/netherworld.dm

```diff
@@ -47,10 +76,115 @@
 	..()
 	if(stat)
 		return
-	if(prob(10))
+	if(prob(10) && !hushed)
 		var/chosen_sound = pick(migo_sounds)
 		playsound(src, chosen_sound, 100, TRUE)
 
+/mob/living/simple_animal/hostile/netherworld/migo/Logout()
+	..()
+	hushed = FALSE
+	if(impersonation)
+		ChangeVoice()
+
+/mob/living/simple_animal/hostile/netherworld/migo/GetVoice()
+	if(impersonation)
+		return impersonation
+	return "[src]"
+
+/mob/living/simple_animal/hostile/netherworld/migo/proc/Hush()
+	set name = "Hush"
+	set category = "Nether"
+	set desc = "Toggles passive noises and noises when you talk."
+	if(hushed)
+		hushed = FALSE
+		to_chat(src, "<span class='notice'>You will now occasionally mimic noises in this dimension.</span>")
+	else
+		hushed = TRUE
+		to_chat(src, "<span class='notice'>You decide against mimicking noises for now.</span>")
+
+/mob/living/simple_animal/hostile/netherworld/migo/proc/ChangeVoice()//hey, not returning null in this is important! don't @ me!
+	set name = "Tune Voice"
+	set category = "Nether"
+	set desc = "Changes your voice to someone else. You will copy the voice patterns if the name matches them."
+	if(impersonation)
+		if(client)
+			to_chat(src, "<span class='notice'>You are no longer impersonating [impersonation].</span>")
+		speak_emote = initial(speak_emote)
+		impersonation = null
+		return TRUE
+	else
+		var/t = copytext(sanitize(input(src, "Enter the name of your victim!", "Mimic", name) as text), 1, MAX_NAME_LEN)
+		if(!t || t == "Unknown" || t == "floor" || t == "wall" || t == "r-wall") //Same as mob/dead/new_player/prefrences.dm
```

Comment:
If this is a repeat from there, extract it to some blacklist or name input proc.

Reply:
Sure, where do you think works best?

Reply:
_helpers/names.dm if you want full proc, _globalvars/lists/names.dm if  you want to extract the invalid names only

Reply:
👍

Reply:
tfw you can name yourself as Toxins Scrubber `#`2 but not Wall Ero Bot

Reply:
it would have to exactly be "wall" or other names like that for the code to slap your shit

Reply:
just name yourself reinforced wall lol

Reply:
😠 but the point is to stop them from naming themselves after the actual wall

Reply:
do what grieffers on garry'smod do and replace a l with 1